waspc/src/Wasp/Node/Version.hs

```diff
@@ -39,79 +40,85 @@ data VersionCheckResult
+  npmResult <- checkUserNpmVersion
+  return $ case (nodeResult, npmResult) of
+    (VersionCheckSuccess, VersionCheckSuccess) -> VersionCheckSuccess
+    (VersionCheckFail nodeError, _) -> VersionCheckFail nodeError
+    (_, VersionCheckFail npmError) -> VersionCheckFail npmError
+
+checkUserNodeVersion :: IO VersionCheckResult
+checkUserNodeVersion = checkUserToolVersion "node" ["--version"] oldestWaspSupportedNodeVersion
+
+checkUserNpmVersion :: IO VersionCheckResult
+checkUserNpmVersion = checkUserToolVersion "npm" ["--version"] oldestWaspSupportedNpmVersion
+
+checkUserToolVersion :: String -> [String] -> SV.Version -> IO VersionCheckResult
+checkUserToolVersion commandName commandArgs oldestSupportedToolVersion = do
+  userVersion <- getToolVersionFromCommandOutput commandName commandArgs
+  return $ case userVersion of
+    Left errorMsg -> VersionCheckFail errorMsg
+    Right version
+      | version >= oldestSupportedToolVersion -> VersionCheckSuccess
+      | otherwise -> VersionCheckFail $ makeVersionMismatchErrorMessage version
   where
-    nodeVersionParser = skipAnyCharTillMatch SV.versionParser
-    skipAnyCharTillMatch p = P.manyTill P.anyChar (P.lookAhead $ P.try p) >> p
+    makeVersionMismatchErrorMessage version =
+      unlines
+        [ "Your " ++ commandName ++ " version does not meet Wasp's requirements!",
+          "You are running " ++ commandName ++ " " ++ show version ++ ".",
+          "Wasp requires " ++ commandName ++ " version " ++ show oldestSupportedToolVersion ++ " or higher."
+        ]
+
+getToolVersionFromCommandOutput :: String -> [String] -> IO (Either ErrorMessage SV.Version)
+getToolVersionFromCommandOutput commandName commandArgs = do
+  commandOutput <- readCommandOutput commandName commandArgs
+  return $ commandOutput >>= parseVersionFromCommandOutput
 
-nodeNotFoundMessage :: String
-nodeNotFoundMessage = "`node` command not found!"
+readCommandOutput :: String -> [String] -> IO (Either ErrorMessage String)
+readCommandOutput commandName commandArgs = do
+  commandResult <-
+    catchIOError
+      (Right <$> readProcessWithExitCode commandName commandArgs "")
+      (return . Left . wrapCommandIOErrorMessage . makeIOErrorMessage)
+  return $ case commandResult of
+    Left procErr -> Left procErr
+    Right (ExitFailure exitCode, _, stderr) -> Left $ wrapCommandExitCodeErrorMessage exitCode stderr
+    Right (ExitSuccess, stdout, _) -> Right stdout
+  where
+    makeIOErrorMessage ioErr
+      | isDoesNotExistError ioErr = "`" ++ fullCommand ++ "` command not found!"
+      | otherwise = show ioErr
+
+    wrapCommandIOErrorMessage innerErr =
+      unlines
+        [ "Running `" ++ fullCommand ++ "` failed.",
+          indent 2 innerErr,
+          "Make sure you have `" ++ commandName ++ "` installed and in your PATH."
+        ]
+
+    wrapCommandExitCodeErrorMessage exitCode commandErr =
+      unlines
+        [ "Running `" ++ fullCommand ++ "` failed (exit code " ++ show exitCode ++ "):",
```

Comment:
Does wasp usually use backticks when quoting code/commands in its output? I'm not sure.

Please check the rest of our messages to ensure its consistent.